lib/telemetry_metrics_statsd.ex

```diff
@@ -567,10 +642,41 @@ defmodule TelemetryMetricsStatsd do
   defp update_pool(pool_id, new_host, new_port) do
     pool_id
     |> :ets.tab2list()
-    |> Enum.each(fn {:udp, udp} ->
-      :ets.delete_object(pool_id, {:udp, udp})
-      updated_udp = UDP.update(udp, new_host, new_port)
-      :ets.insert(pool_id, {:udp, updated_udp})
+    |> Enum.each(fn item ->
+      case item do
+        {:udp, udp} ->
+          :ets.delete_object(pool_id, {:udp, udp})
+          updated_udp = UDP.update(udp, new_host, new_port)
+          :ets.insert(pool_id, {:udp, updated_udp})
+
+        _ ->
+          :ok
+      end
+    end)
+  end
+
+  def close_udps(pid) do
```

Comment:
Is this function only for test purposes?

Reply:
yes, it is

Reply:
yes, this one accesses private state of genserver, and part of interface, improved the name a bit and added a comment